game/scripts/vscripts/events.lua

```diff
@@ -409,6 +410,9 @@ function GameMode:OnItemCombined(keys)
 
   -- The cost of the item purchased
   local itemcost = keys.itemcost
+
+  local hero = player:GetAssignedHero()
+  local hthinker = CreateModifierThinker( hero, hero , "modifier_aura_item_upgrade", { ItemName = itemName, PlayerId = plyID}, hero:GetOrigin(), hero:GetTeamNumber(), false )
```

Comment:
I'm not sure why you're using a modifier thinker in the first place. Also, the second argument is the ability the modifier is attached to, so you shouldn't be putting `hero` there. Use `nil` if there's not related ability.